mlvm/mlvm.sh

```diff
@@ -0,0 +1,217 @@
+#!/bin/bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This initialization action will download a set of frequently-used Machine Learning
+# libraries onto a Dataproc cluster, as well as GPU support and connectors for
+# Google Cloud Storage, BigQuery and Spark-Bigquery. 
+
+set -euxo pipefail
+
+readonly JARS_DIR=/usr/lib/spark/jars
+readonly CONNECTORS_DIR=/usr/local/share/google/dataproc/lib
+
+readonly DEFAULT_INIT_ACTIONS_REPO=gs://dataproc-initialization-actions
+readonly INIT_ACTIONS_REPO="$(/usr/share/google/get_metadata_value attributes/init-actions-repo ||
+  echo ${DEFAULT_INIT_ACTIONS_REPO})"
+readonly INIT_ACTIONS_DIR=$(mktemp -d -t dataproc-init-actions-XXXX)
+
+readonly INCLUDE_GPUS="$(/usr/share/google/get_metadata_value attributes/include-gpus || true)"
+readonly SPARK_BIGQUERY_VERSION="$(/usr/share/google/get_metadata_value attributes/spark-bigquery-connector-version ||
+  echo "0.17.0")"
+
+readonly R_VERSION_ENV="$(R --version | sed -n 's/.*version[[:blank:]]\+\([0-9]\+\.[0-9]\).*/\1/p' )"
+
+if [[ ${R_VERSION_ENV} == "4"* ]]; then
+  R_VERSION="4.0"
+else
+  R_VERSION="3.6.0"
+fi
+
+# Pip
+BASE_PYTHON_PACKAGES=(
+  "google-cloud-bigquery==1.26.1" 
+  "google-cloud-datalabeling==0.4.0"
+  "google-cloud-storage==1.30.0"
+  "google-cloud-bigtable==1.4.0" 
+  "google-cloud-dataproc==1.0.1" 
+  "google-api-python-client==1.10.0" 
+  "matplotlib==3.3.0"
+  "mxnet==1.6.0" 
+  "nltk==3.5"
+  "numpy==1.18.4" 
+  "rpy2==3.3.3"
+  "scikit-learn==0.23.1" 
+  "scipy==1.4.1"
+  "sparksql-magic==0.0.3" 
+  "tensorflow-datasets==3.2.1"
+  "tensorflow-estimator==2.2.0"
+  "tensorflow-hub==0.8.0"
+  "tensorflow-io==0.14.0"
+  "tensorflow-probability==0.10.1" 
+  "torch==1.5.1" 
+  "torchvision==0.6.1" 
+  "xgboost==1.1.0"
+)
```

Comment:
What will happen if we will not pin versions? Or maybe pin just couple top-level packages? It could be toilsome if we need to update/maintain them manually?

Reply:
Just a potential issue with unexpected version conflicts / breaking changes. I think it's good to provide the versioning transparently to the user as well. Going forward maybe we can use https://github.com/renovatebot/renovate.

Reply:
We don't do this even for base Dataproc image and rely on Ana/Miniconda version to resolve dependencies. The problem that it should work for future Dataproc versions but we do not version init actions, so we need to support this map per-image track or just relay on conda version to resolve dependencies.

Reply:
Ok, in this case it probably makes sense to install these via conda where appropriate / possible. I'll look into.